src/additional_functions/helper.jl

```diff
@@ -114,11 +114,11 @@ function sparse_outer_mul!(C, A, B::Vector, ind) #computes A*S*B -> C, where ind
 end
 
 function cov_and_mean(rows; corrected = false)
-    data = permutedims(hcat(rows...))
+    data = mapreduce(transpose, vcat, rows)
```

Comment:
When I benchmark it, this version is quite a bit faster:
```suggestion
    data = transpose(reduce(hcat, rows))
```

Reply:
Interesting. I have thought too your version would be faster, but on my system (julia 1.10.1) the mapreduce was a bit faster and allocated less (by 25%). But if that's something that was fixed in julia just recently, I guess it is better to use your suggestion.

Reply:
Hm, I was on 1.9, I now tried on

> Julia Version 1.10.2
Commit bd47eca2c8a (2024-03-01 10:14 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Linux (x86_64-linux-gnu)
  CPU: 12 × 12th Gen Intel(R) Core(TM) i7-1265U
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, alderlake)
Threads: 1 default, 0 interactive, 1 GC (on 12 virtual cores)

getting the same results (`transpose(reduce(hcat, rows))`) being faster by a lot). So maybe it's something else (multithreading/os maybe). But as you said, since the difference is not that large for you it should be fine to go ahead with `transpose(reduce(hcat, rows))` (both will be an improvement to before^^).